pkgs/by-name/xr/xrootd/package.nix

```diff
@@ -0,0 +1,162 @@
+{
+  lib,
+  stdenv,
+  callPackage,
+  fetchFromGitHub,
+  davix,
+  cmake,
+  gtest,
+  makeWrapper,
+  pkg-config,
+  curl,
+  isa-l,
+  fuse,
+  libkrb5,
+  libuuid,
+  libxcrypt,
+  libxml2,
+  openssl,
+  readline,
+  scitokens-cpp,
+  systemd,
+  voms,
+  zlib,
+  # If not null, the builder will
+  # move "$out/etc" to "$out/etc.orig" and symlink "$out/etc" to externalEtc.
+  externalEtc ? "/etc",
+  removeReferencesTo,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "xrootd";
+  version = "5.7.1";
+
+  src = fetchFromGitHub {
+    owner = "xrootd";
+    repo = "xrootd";
+    rev = "refs/tags/v${finalAttrs.version}";
+    fetchSubmodules = true;
+    hash = "sha256-ZU31nsQgs+Gz9mV8LVv4utJ7g8TXN5OxHjNDfQlt38M=";
+  };
+
+  outputs = [
+    "bin"
+    "out"
+    "dev"
+    "man"
+  ] ++ lib.optional (externalEtc != null) "etc";
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    pkg-config
+    removeReferencesTo
+  ];
+
+  buildInputs =
+    [
+      davix
+      curl
+      isa-l
+      libkrb5
+      libuuid
+      libxcrypt
+      libxml2
+      openssl
+      readline
+      scitokens-cpp
+      zlib
+    ]
+    ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [
+      # https://github.com/xrootd/xrootd/blob/5b5a1f6957def2816b77ec773c7e1bfb3f1cfc5b/cmake/XRootDFindLibs.cmake#L58
+      fuse
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      systemd
+      voms
+    ];
+
+  preConfigure =
```

Comment:
Why is that preConfigure and not postPatch?

Reply:
Changed it.